envoy/cilium/npds.proto

```diff
@@ -32,7 +32,7 @@ service NetworkPolicyDiscoveryService {
 message NetworkPolicy {
   // The unique identifier of the network policy.
   // Required.
-  uint64 policy = 1;
+  string name = 1;
```

Comment:
Is this safe? I suppose we ensure our envoy is in sync with the agent but we introduced some support for shared envoys recently. I don't expect any of our code is sensitive to field order or field number (if so, that seems ill advised). Could you make name the new #4 field? (assuming I'm correct about proto3 expectations).

Reply:
For now, we ensure that the envoy and cilium versions match down to the commit SHA, so we're free to change the protobuf definition.

In future this will definitely be a concern.

Reply:
This protocol is not yet used by anything, so we can still change it in a non-backward-compatible way.